File: contracts/aaa/aaa.go
// Copyright 2020 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

package aaa

import "github.com/iotaledger/wasp/packages/vm/wasmlib"


func funcMyFunc(ctx *wasmlib.ScFuncContext, params *FuncMyFuncParams) {
    ctx.Log("calling myFunc")
}

func viewMyView(ctx *wasmlib.ScViewContext, params *ViewMyViewParams) {
    ctx.Log("calling myView")
}

File: contracts/aaa/src/aaa.rs
// Copyright 2020 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

use wasmlib::*;

use crate::*;

pub fn func_my_func(ctx: &ScFuncContext, params: &FuncMyFuncParams) {
    ctx.log("calling myFunc");
}

pub fn view_my_view(ctx: &ScViewContext, params: &ViewMyViewParams) {
    ctx.log("calling myView");
}
